.github/actions/build-push-image/action.yaml

```diff
@@ -88,34 +82,65 @@ runs:
       shell: bash
       run: |
         go mod download
-    - name: Build and Push Operator to Docker Registry
+    - name: Build all platforms
+      shell: bash
+      run: make all-platforms
+    - name: Fail if send input flags are both unset
+      if: ${{ inputs.push_to_docker != 'true' && inputs.push_to_quay != 'true' }}
+      shell: bash
+      run: exit 1
+    - name: Compute main build tag
+      id: tag
+      shell: bash
+      run: |
+        main_tag=${{ inputs.push_to_docker == 'true' && format('{0}:{1}', inputs.repository, inputs.version) || format('quay.io/{0}:{1}', inputs.repository, inputs.version)}}
+        echo "main_tag=$main_tag" >> $GITHUB_OUTPUT
+    - name: Login to docker registry
       if: ${{ inputs.push_to_docker == 'true' }}
-      uses: docker/build-push-action@v3
+      uses: docker/login-action@v2
       with:
-        context: .
-        file: ${{ inputs.file }}
-        build-args: VERSION=${{ inputs.version }}
-        platforms: ${{ inputs.platforms }}
-        tags: ${{ inputs.repository }}:${{ inputs.version }}
-        cache-from: type=gha
-        cache-to: type=gha,mode=max
-        push: true
+        username: ${{ inputs.docker_username }}
+        password: ${{ inputs.docker_password }}
     - name: Login to quay.io registry
       if: ${{ inputs.push_to_quay == 'true' }}
       uses: docker/login-action@v2
       with:
         registry: quay.io
         username: ${{ inputs.quay_username }}
         password: ${{ inputs.quay_password }}
-    - name: Build and Push Operator to Quay Registry
-      if: ${{ inputs.push_to_quay == 'true' }}
+    - name: Build and Push Operator to ${{ steps.tag.outputs.main_tag }}
       uses: docker/build-push-action@v3
       with:
         context: .
         file: ${{ inputs.file }}
         build-args: VERSION=${{ inputs.version }}
         platforms: ${{ inputs.platforms }}
-        tags: quay.io/${{ inputs.repository }}:${{ inputs.version }}
         cache-from: type=gha
         cache-to: type=gha,mode=max
+        sbom: true
         push: true
+        tags: ${{ steps.tag.outputs.main_tag }}
+    - name: Push To Quay
+      if: ${{ inputs.push_to_quay == 'true' && inputs.push_to_docker == 'true' }}
+      shell: bash
+      run: |
+        docker pull ${{ steps.tag.outputs.main_tag }}
+        docker tag ${{ steps.tag.outputs.main_tag }} quay.io/${{ inputs.repository }}:${{ inputs.version }}
+        echo "PUSH: docker push quay.io/${{ inputs.repository }}:${{ inputs.version }}"
+        docker push quay.io/${{ inputs.repository }}:${{ inputs.version }}
+    - name: Push Certified to Docker Registry
```

Comment:
Certified image would never be pulled from docker, it doesn't make sense to actually push to it. I'd remove this step

Reply:
will fix

Reply:
Done